integrations/FacebookPixel/browser.js

```diff
@@ -76,10 +77,30 @@ class FacebookPixel {
 
   identify(rudderElement) {
     if (this.advancedMapping) {
+      const reserve = [
+        "email",
+        "lastName",
+        "firstName",
+        "phone",
+        "external_id",
+        "city",
+        "birthday",
+        "gender",
+        "street",
+        "zip",
+        "country",
+      ];
+      // this construcPayload will help to map the traits in the same way as cloud mode
+      let payload = constructPayload(rudderElement.message, traitsMapper);
       const traits = rudderElement.message.context
         ? rudderElement.message.context.traits
         : undefined;
-      window.fbq("init", this.pixelId, traits);
+      // here we are sending other traits apart from the reserved ones.
+      reserve.forEach((element) => {
+        delete traits[element];
+      });
+      payload = { ...payload, ...traits };
+      window.fbq("init", this.pixelId, payload);
```

Comment:
With these changes, the final payload looks quite different than before. Wouldn't this be a breaking change for users?
CC @utsabc

Reply:
I don't think there is any use of other attributes apart from [these](https://developers.facebook.com/docs/meta-pixel/advanced/advanced-matching).

Reply:
The sample payloads sent to FB SDK are as follows:
Before:
```
{
  "firstName": "John",
  "lastName": "Cena",
  "email": "john@wwe.com",
  "phone": "",
  "gender": "m",
  "birthday": "19990101",
  "city": "London",
  "country": "England",
  "zip": "81929120",
  "foo": "bar"
}
```

After:
```
{
  "external_id": "user001",
  "em": "john@wwe.com",
  "ph": "",
  "ge": "m",
  "db": "19990101",
  "ln": "Cena",
  "fn": "John",
  "ct": "London",
  "zp": "81929120",
  "country": "England",
  "foo": "bar"
}
```

Clearly, the field names are quite different and there is a good possibility that the users are affected if they were relying on the previously sent fields.
We need to make sure that the changes are backward compatible.
CC @utsabc

Reply:
@ItsSudip We should create a flag for backward compatibility. `Use Updated Mapping` by default it should be kept false also we should say that we recommend migrating as the earlier one will be depreciating

Reply:
@ItsSudip We should keep a config in the web app for backward compatibility. For example `use updated mapping` which should be false by default when enabled the above mapping will be triggered.
Also let's inform in a callout that we will depreciate the existing mapping soon